### PR TITLE
Music panel

### PR DESCRIPTION
**Pull request by CurtisGreen**
image

- Added the music panel
- Songs change color from red -> green when you unlock them
- Current song is displayed
- Clicking on an unlocked song will play that song until you move to another area
- Currently unlocked / total songs updates at the bottom

Note:
I didn't add the functionality for auto/man/loop

### CODE REVIEW

File: src/ui/music.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        this.updateSongUnlocks();
+    }
+
+    updateSongUnlocks() {
+        // Update song name color
+        this.songTexts.forEach((textObj) => {
+            if (!textObj.unlocked && characterData.getQuestCompleted(textObj.prereq)) {
+                textObj.unlocked = true;
+                textObj.setStyle(FONTS.SONG_UNLOCKED);
+                console.log("unlocked", textObj.text);
```
Comment (EthanVieira) [inline src/ui/music.js]: Thoughts on logging something to chat window on song unlock?

File: src/ui/music.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            }
+        });
+
+        // Update current / total unlocked
+        if (this.numUnlockedText != undefined) {
```
Comment (EthanVieira) [inline src/ui/music.js:151]: Thoughts on adding a congrats log to chat window when all of the songs are unlocked?

**Review by EthanVieira — CHANGES_REQUESTED** [review]
Nice work on this - looks really good. Only left two comments about logging.

File: src/ui/music.js (hunk 1 above)
Comment (CurtisGreen) [reply src/ui/music.js]: Sounds good to me

File: src/ui/music.js (hunk 2 above)
Comment (CurtisGreen) [reply src/ui/music.js:151]: Done https://github.com/EthanVieira/rs-clicker/pull/53/commits/3f36a73bc262e3b3a2cb541cbbbdfc4e65822ed3

File: src/ui/music.js (hunk 1 above)
Comment (CurtisGreen) [reply src/ui/music.js]: Done https://github.com/EthanVieira/rs-clicker/pull/53/commits/3f36a73bc262e3b3a2cb541cbbbdfc4e65822ed3

**Review by EthanVieira — APPROVED** [review]

**Pull request merged**